### ISSUE TITLE
LightGBMTuner don't seem to be reproducible

### ISSUE THREAD

**Issue by RegiusQuant** [optuna/optuna#1793]
## Expected behavior

I use LightGBMTuner to find params, but each run on the same dataset has different results

## Environment

- Optuna version: 2.1.0
- Python version: 3.8.5
- OS: Linux
- (Optional) Other libraries and their versions:

## Error messages, stack traces, or logs

```
# error messages, stack traces, or logs
```

## Steps to reproduce

1.
2.
3.

## Reproducible examples (optional)

```python
# python code
```

**Comment by HideakiImamura** [comment]
Thanks for the report! Actually, `optuna.integration.lightgbm.LightGBMTuner` does not have any `seed` options, but `optuna.integration.lightgbm.LightGBMTunerCV` have `seed` option. I think we should add `seed` option to the `LightGBMTuner`.

@toshihikoyanase I'd like to know your opinion. What do you think?

**Comment by toshihikoyanase** [comment]
@HideakiImamura Thank you for letting me know.
We received a similar bug report (https://github.com/optuna/optuna/issues/1740#issuecomment-682494087), and I agree with you.

## Seed parameters in Lightgbm tuner

We will have three kinds of seeds in the LightGBM tuner.

1. A seed option in `params` of `lightgbm`
1. A seed as an argument of `lightgbm.cv`
1. A seed for Optuna samplers

The first `seed` is used by both `lightgbm.train` and `lightgbm.cv`, but the second seed is only used by `lightgbm.cv` to split the CV folds.

## Design choices

I think we have two design choices to implement the seed option.

1. Reuse the `seed` option in `params`

Since the original lightgbm has multiple seeds, it may be confusing if we add the seed option for Optuna. If we reuse the `seed` in `params`, users will not have to take care of Optuna's seeding. On the other hand, they may want to change the Optuna seed while keeping the seed for `lightgbm` to check the optimization performance/stability of Optuna.

2. Add `optuna_seed` argument to `LightGBMTuner.__init__` and `LightGBMTunerCV.__init__`

This design is more flexible, and it can keep backward compatibility. If users accept additional complexity, this choice may be better than the previous one.

**Comment by tetsuoh0103** [comment]
@toshihikoyanase 
Hi, I was faced with this issue and created sample fix based on version==2.5.0.
The design is basically same with 2. in the above discussion.
Could you consider to take a look if this is what you expected. If OK, I will create PR.
[fix_sample.zip](https://github.com/optuna/optuna/files/6061903/fix_sample.zip)

**Comment by toshihikoyanase** [comment]
@tetsuoh0103 Thank you for your suggestion. Your code will be a great help to us.
In general, sharing the code in zip archives has potential security risks like unintended contamination of malicious code, so could you directly send a draft PR, please?

**Comment by tetsuoh0103** [comment]
@toshihikoyanase  Thank you for your comment and sorry for the inconvenience. Please take a look at the PR#2431.

**Comment by toshihikoyanase** [comment]
@tetsuoh0103 Thank you for your PR. I'll check it.

**Comment by z7ye** [comment]
I also saw the same issue with Xgboost.

**Comment by nzw0301** [comment]
hi @RegiusQuant, I thinl #2431 might fix the issue by introducing seed parameter for optuna sampler. What do you think about it?

**Comment by Crissman** [comment]
Since the original issue of making LightGBMTuner reproducible was accomplished with PR #2431, I'm closing this issue.

If there are additional areas like XGBoost that should be made reproducible, please open another issue.

Thanks,
